### PR TITLE
Add constraints file during installation of packages from pip in docs workflow

### PR DESCRIPTION
**Pull request by Czaki**
# Description

Add usage of constraints in build docs process to reduce failed workflows caused by updates of dependency. 

I also add a lower pin to typing-extension to keep min-req test working. 

# References

## Type of change

- [x] Bug-fix (non-breaking change which fixes an issue)

# How has this been tested?

- [ ] example: the test suite for my feature covers cases x, y, and z
- [ ] example: all tests pass with my change
- [ ] example: I check if my changes works with both PySide and PyQt backends
      as there are small differences between the two Qt bindings.  

## Final checklist:
- [ ] My PR is the minimum possible work for the desired functionality
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If I included new strings, I have used `trans.` to make them localizable.
      For more information see our [translations guide](https://napari.org/developers/translations.html).

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -76,7 +76,7 @@ install_requires =
     tifffile>=2020.2.16
     toolz>=0.10.0
     tqdm>=4.56.0
-    typing_extensions
+    typing_extensions>=4.2.0
```
Comment (jni) [inline setup.cfg:79]: why is this change required in addition to the constraints?

**Review by jni — APPROVED** [review]
@Czaki approved but just a small question about the setup.cfg changes. Maybe that should be a different PR?

File: setup.cfg (hunk 1 above)
Comment (Czaki) [reply setup.cfg:79]: Because without this in min req test the latest typing extension is insstaled. And old pydantic is incompatible with the latest typing extension

Comment (Czaki) [reply setup.cfg:79]: I could split it to separate PR, but I want to reduce the time to merge bugfix as fast as possible.

**Comment by psobolewskiPhD** [comment]
@Czaki docs are building here in this PR, but not in napari/docs -- do we need to port this to that repo too?

**Comment by Czaki** [comment]
I think yes.

**Comment by psobolewskiPhD** [comment]
I merged main and looks like the docs issue is fixed for PRs in this repo:
https://github.com/napari/napari/actions/runs/5076767489/jobs/9119271941?pr=5618
But build docs action in PRs in docs repo still fail:
https://github.com/napari/docs/actions/runs/5059087523/jobs/9119586997
 so looks like we need to port over this fix (and I guess it means some dependency update is crashing our examples?)
